Asterix/optics/optical_systems.py

```diff
@@ -36,6 +36,13 @@ def __init__(self, modelconfig):
         if modelconfig["diam_pup_in_pix"] % 2 == 1:
             raise ValueError("Please set [modelconfig]['diam_pup_in_pix'] parameter to an even number.")
 
+        if "complex_precision" in modelconfig.keys():
```

Comment:
When querying a dict, it queries the keys by default, so you don't need the `.keys()` here.

Reply:
done